app/utils/skills-list.js

```diff
@@ -0,0 +1,48 @@
+import Ember from 'ember';
+
+const {
+  get
+} = Ember;
+
+let skillsList = {
+  find(skills, skill, relationship) {
+    if (skills) {
+      let userSkill = skills.find((item) => {
+        let model = get(item, 'constructor.modelName');
+        return _matchForModel(model, item, skill, relationship);
+      });
+      return userSkill;
+    }
+  },
+
+  contains(skills, skill) {
+    if (skills) {
+      return skills.any((item) => {
+        let itemSkillId = item.belongsTo('skill').id();
+        let skillId = get(skill, 'id');
+        return (itemSkillId === skillId);
+      });
+    } else {
+      false;
+    }
+  }
+};
+
+function _matchForModel(model, item, skill, relationship) {
+  switch (model) {
+    case 'user-skill':
+      return _matchWithRelationship(item, skill, relationship, 'user');
```

Comment:
@JoshSmith I am not sure what the plan is down the line, or if the other model types would leverage the same `_matchWithRelationship` function. If they did you could do a mapping object for relationship name. Otherwise, this seems like a valid use-case for a `switch` statement.